elide-async/src/main/java/com/yahoo/elide/async/service/AsyncQueryThread.java

```diff
@@ -0,0 +1,140 @@
+import javax.ws.rs.core.MultivaluedMap;
+
+import org.apache.http.NameValuePair;
+import org.apache.http.client.utils.URIBuilder;
+
+import com.yahoo.elide.Elide;
+import com.yahoo.elide.ElideResponse;
+import com.yahoo.elide.async.models.AsyncQuery;
+import com.yahoo.elide.async.models.AsyncQueryResult;
+import com.yahoo.elide.async.models.QueryStatus;
+import com.yahoo.elide.async.models.QueryType;
+import com.yahoo.elide.graphql.QueryRunner;
+import com.yahoo.elide.security.User;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Runnable thread for executing the query provided in Async Query.
+ * It will also update the query status and result object at different
+ * stages of execution.
+ */
+@Slf4j
+@Data
+@AllArgsConstructor
+public class AsyncQueryThread implements Runnable {
+
+    private String query;
+    private QueryType queryType;
+    private User user;
+    private Elide elide;
+    private QueryRunner runner;
+    private UUID id;
+
+    @Override
+    public void run() {
+        processQuery();
+    }
+
+    /**
+     * This is the main method which processes the Async Query request, executes the query and updates
+     * values for AsyncQuery and AsyncQueryResult models accordingly.
+     */
+    protected void processQuery() {
+        AsyncDbUtil asyncDbUtil = AsyncDbUtil.getInstance(elide);
+        try {
+            // Change async query to processing
+            asyncDbUtil.updateAsyncQuery(id, (asyncQuery) -> {
+                asyncQuery.setStatus(QueryStatus.PROCESSING);
+                });
+            //Thread.sleep(180000);
+            ElideResponse response = null;
+            log.debug("query: {}", query);
+            log.debug("queryType: {}", queryType);
+            AsyncQuery asyncQuery;
+            AsyncQueryResult asyncQueryResult;
+            if (queryType.equals(QueryType.JSONAPI_V1_0)) {
+                MultivaluedMap<String, String> queryParams = getQueryParams(query);
+                response = elide.get(getPath(query), queryParams, user);
+                log.debug("JSONAPI_V1_0 getResponseCode: {}", response.getResponseCode());
+                log.debug("JSONAPI_V1_0 getBody: {}", response.getBody());
+            }
+            else if (queryType.equals(QueryType.GRAPHQL_V1_0)) {
+                response = runner.run(query, user);
+                log.debug("GRAPHQL_V1_0 getResponseCode: {}", response.getResponseCode());
+                log.debug("GRAPHQL_V1_0 getBody: {}", response.getBody());
+            }
+            if (response != null){
+                asyncQuery = asyncDbUtil.updateAsyncQuery(id, (asyncQueryObj) -> {
+                    asyncQueryObj.setStatus(QueryStatus.COMPLETE);
+                });
+
+                // Create AsyncQueryResult entry for AsyncQuery
+                asyncQueryResult = asyncDbUtil.createAsyncQueryResult(response.getResponseCode(), response.getBody(), asyncQuery, id);
```

Comment:
Creating the response and updating the status should be done inside a single transaction to the database.